### PR TITLE
add test for get_case_properties

### PR DESCRIPTION
**Pull request by dannyroberts**
- add utility create_real_cases_from_dummy_cases for making actual cases
  from unsaved CommCareCase objects

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/util.py [hunk 1]
```diff
@@ -43,7 +48,34 @@ def post_case_blocks(case_blocks, form_extras=None, domain=None):
         **form_extras
     )
     response, xform, cases = sp.run()
-    return xform
+    return xform, cases
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/case/util.py]: I'm kind of surprised there weren't more references to this to change (though I assume you looked).

Not sure if it feels out of scope, but you could have `CaseFactory` do fewer DB calls by tweaking this function to leverage the new API and eliminate this call: https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/casexml/apps/case/mock.py#L377

Comment (dannyroberts) [inline corehq/ex-submodules/casexml/apps/case/util.py]: Yeah I was a little surprised too, but I do remember bumping into this last time I refactored this function; all of the other calls to it don't use the return value.

Nice catch on `CaseFactory`. That's only used in tests (and consequently very well tested as well) so I feel fine making that change and relying on the tests. It's simpler and may have the side effect of speeding up the tests a tiny bit too (though I wouldn't count on it being noticeable).

**Comment by dannyroberts** [comment]
(ok apparently i missed one place, which I just fixed. Hopefully only one)

**Comment by dannyroberts** [comment]
:white_check_mark:

**Comment by dannyroberts** [comment]
@kaapstorm buddy. @czue

**Comment by czue** [comment]
confirming you aren't going to make the `CaseFactory` changes (or are doing those separately).

**Comment by dannyroberts** [comment]
Got that ready to PR right here: https://github.com/dimagi/commcare-hq/compare/CaseFactory

**Pull request merged**